test/GPv2TradeExecution.test.ts

```diff
@@ -0,0 +1,147 @@
+import IERC20 from "@openzeppelin/contracts/build/contracts/IERC20.json";
+import { expect } from "chai";
+import { Contract, BigNumber } from "ethers";
+import { artifacts, ethers, waffle } from "hardhat";
+
+export interface ExecutedTrade {
+  owner: string;
+  sellToken: string;
+  buyToken: string;
+  sellAmount: BigNumber;
+  buyAmount: BigNumber;
+}
+
+export function composeExecutedTrade(
+  trade: ExecutedTrade,
+): [string, string, string, BigNumber, BigNumber] {
+  return [
+    trade.owner,
+    trade.sellToken,
+    trade.buyToken,
+    trade.sellAmount,
+    trade.buyAmount,
+  ];
+}
+
+describe("GPv2TradeExecution", () => {
+  const [deployer, recipient, ...traders] = waffle.provider.getWallets();
+
+  let tradeExecution: Contract;
+
+  beforeEach(async () => {
+    const GPv2TradeExecution = await ethers.getContractFactory(
+      "GPv2TradeExecutionTestInterface",
+      recipient,
+    );
+
+    tradeExecution = await GPv2TradeExecution.deploy();
+  });
+
+  describe("transferSellAmountToRecipient", () => {
+    const withoutBuy = {
+      buyToken: ethers.constants.AddressZero,
+      buyAmount: ethers.constants.Zero,
+    };
+
+    it("should transfer sell amount to recipient", async () => {
+      const amount = ethers.utils.parseEther("13.37");
+
+      const sellToken = await waffle.deployMockContract(deployer, IERC20.abi);
+      await sellToken.mock.transferFrom
+        .withArgs(traders[0].address, recipient.address, amount)
+        .returns(true);
+
+      await expect(
+        tradeExecution.transferSellAmountToRecipientTest(
+          composeExecutedTrade({
+            owner: traders[0].address,
+            sellToken: sellToken.address,
+            sellAmount: amount,
+            ...withoutBuy,
+          }),
+          recipient.address,
+        ),
+      ).to.not.be.reverted;
+    });
+
+    it("should revert on failed ERC20 transfers", async () => {
+      const amount = ethers.utils.parseEther("4.2");
+
+      const sellToken = await waffle.deployMockContract(deployer, IERC20.abi);
+      await sellToken.mock.transferFrom
+        .withArgs(traders[0].address, recipient.address, amount)
+        .revertsWithReason("test error");
+
+      await expect(
+        tradeExecution.transferSellAmountToRecipientTest(
+          composeExecutedTrade({
+            owner: traders[0].address,
+            sellToken: sellToken.address,
+            sellAmount: amount,
+            ...withoutBuy,
+          }),
+          recipient.address,
+        ),
+      ).to.be.revertedWith("test error");
+    });
+
+    it("should revert when transfering from an address without code", async () => {
```

Comment:
Very nitpicky, but I was a bit confused.
```suggestion
    it("should revert when transfering a token with no contract at its address", async () => {
```